Aditi dataset

AditiBansal711 opened the pull request:
<img width="1310" height="1296" alt="image" src="https://github.com/user-attachments/assets/76f3b249-33df-4e9d-9ab4-fa2a814e9366" />

Login screen ready, working on dataset and how it will be placed still!

AditiBansal711 committed:
Connect MenuScreenController to main.ts and add model documentation

AditiBansal711 committed:
Update types.ts

AditiBansal711 committed:
Run Prettier formatting

AditiBansal711 committed:
Fix Prettier formatting in MenuScreenController

AditiBansal711 committed:
login screen added

AditiBansal711 committed:
resume works

AditiBansal711 committed:
Merge branch 'main' into aditi-dataset

7174Andy reviewed (changes requested):
Before requesting a review, can you please rebase main? Thank you!

AditiBansal711 committed:
Add SQLite database integration and fix resume button functionality

- Integrate SQLite database using sql.js for user data storage
- Implement resume button to restore user's last screen after login
- Fix font usage: replace FONT_FAMILY with FONTS.topic in LoginScreenView
- Fix MenuScreenController bug: pass Screen object instead of string to setLastScreen
- Remove unused imports and variables (Screen, resumeEnabled, config)
- Update LoginScreenModel and Controller to handle async database operations
- Initialize SQLite database on app startup

AditiBansal711 committed:
Fix login and resume button functionality

- Fix login button to navigate to map view after login
- Fix resume button to skip login and go directly to last screen
- Make database initialization non-blocking so app starts immediately
- Add debugging logs for troubleshooting
- Fix menu screen to use controller.show() method
- Ensure buttons are properly listening for clicks

AditiBansal711 committed:
Merge branch 'main' into aditi-dataset

AditiBansal711 committed:
Fix linter formatting issues

AditiBansal711 committed:
Fix: Remove duplicate Minigame1SimulController import

beadibe reviewed (approved):
LGTM

7174Andy reviewed (approved):
LGTM

ndimaya reviewed (approved):
looks good
